tests/unit/Snak/SnakListTest.php

```diff
@@ -294,6 +295,50 @@ public function testOrderByProperty( SnakList $snakList, SnakList $expected, arr
 		}
 	}
 
+	public function testComparableInterface() {
+		$this->assertInstanceOf( Comparable::class, new SnakList() );
+	}
+
+	/**
+	 * @dataProvider equalsProvider
+	 */
+	public function testEquals( SnakList $list1, SnakList $list2, $expected ) {
+		$this->assertSame( $expected, $list1->equals( $list2 ) );
+	}
+
+	public function equalsProvider() {
+		$empty = new SnakList();
+		$oneSnak = new SnakList( [ new PropertyNoValueSnak( 1 ) ] );
+
+		return [
+			'empty object is equal to itself' => [
+				$empty,
+				$empty,
+				true
+			],
+			'non-empty object is equal to itself' => [
+				$oneSnak,
+				$oneSnak,
+				true
+			],
+			'different empty objects are equal' => [
+				$empty,
+				new SnakList(),
+				true
+			],
+			'different objects with same content are equal' => [
+				$oneSnak,
+				new SnakList( [ new PropertyNoValueSnak( 1 ) ] ),
+				true
+			],
+			'different objects with different content are not equal' => [
+				$oneSnak,
+				new SnakList( [ new PropertyNoValueSnak( 2 ) ] ),
+				false
```

Comment:
Both are valid approaches, and even if I agree that splitting test cases is the better approach in many cases, I disagree in this case. Readability is key here, and the names of these test cases almost read like a poem. This poem will be destroyed if the last one is not an array key any more but a method name. I don't think this is worth it in this case, if the only benefit is that five booleans can be removed.